tools/refguide_check.py

```diff
@@ -809,6 +856,48 @@ def check_doctests_testfile(fname, verbose, ns=None,
 
     return results
 
+def iter_included_files(base_path, verbose=0, suffixes=('.py', '.rst')):
+    if os.path.exists(base_path) and os.path.isfile(base_path):
+        yield base_path
+    for dir_name, subdirs, files in os.walk(base_path, topdown=True):
+        if dir_name in RST_SKIPLIST:
+            if verbose > 0:
+                sys.stderr.write('skipping files in %s' % dir_name)
+            files = []
+        for p in RST_SKIPLIST:
+            if p in subdirs:
+                if verbose > 0:
+                    sys.stderr.write('skipping %s and subdirs' % p)
+                subdirs.remove(p)
+        for f in files:
+            if (os.path.splitext(f)[1] in suffixes and
+                    f not in RST_SKIPLIST):
+                yield os.path.join(dir_name, f)
+
+def check_documentation(base_path, results, args, dots):
+    """Check examples in any *.rst or *.py files located inside `base_path`.
+    Add the output to `results`. As opposed to ``check_doctests``, does not
+    inspect the classes' and functions' docstrings
```

Comment:
Does this mean that module-level docstrings are checked twice?

Reply:
how about "does not inspect python object docstrings" ?

Reply:
The purpose of this function is pretty clear already - what I'm more confused by is whether `check_doctests` also checks the module-level docstrings

Reply:
Would it be accurate to say the following?
```
For python files, this checks only module-level docstrings, unlike ``check_doctests`` which checks only docstrings for classes and functions
```

Reply:
the code in `check_doctests_testfile` parses doctests out of the text file it is given without any eval. If that text happens to be the `__doc__` of a class, method, module or function, it will be discovered and tested. For now perhaps it would be easier to just leave out the `*.py` files and only check `*.rst` files.

Reply:
done